src/language-js/printer-estree.js

```diff
@@ -672,8 +683,9 @@ function printPathNoParens(path, options, print, args) {
             options.locEnd
           );
           return (
+            nextCharacter &&
```

Comment:
```suggestion
            nextCharacter !== false &&
```

I wish we had a good way to test this kind of case.